src/agent/restart_handler/src/restart_handler_win.cpp

```diff
@@ -0,0 +1,17 @@
+#include <boost/asio/awaitable.hpp>
+#include <logger.hpp>
+#include <restart_handler.hpp>
+
+namespace restart_handler
+{
+
+    std::vector<char*> RestartHandler::cmd_line;
+
+    boost::asio::awaitable<module_command::CommandExecutionResult> RestartHandler::RestartCommand()
+    {
+        // TODO
```

Comment:
Is there an open issue to implement this?

Reply:
Not yet. Do you think I should create it?

Reply:
Yes, let's open one for Windows and one for macOS.

Reply:
* [Add support for agent self-restarting for Windows #552](https://github.com/wazuh/wazuh-agent/issues/552)
* [Add support for agent self-restarting for macOS #553](https://github.com/wazuh/wazuh-agent/issues/553)